### PR TITLE
[improvement] Add two parameters in workergroup, and support the application of description display and other parameters

### LINKED ISSUE #11546: [Improvement][Module Name] The default worker grouping supports editing
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

The default worker group cannot be edited, and all worker groups have no description field

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Zephyrcf**
## Purpose of the pull request

close https://github.com/apache/dolphinscheduler/issues/11546

## Brief change log
Support the default worker group for editing, and add all worker group description functions.

### CODE REVIEW

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/WorkerGroup.java [hunk 1]
```diff
@@ -92,16 +94,24 @@ public void setSystemDefault(boolean systemDefault) {
...
-                + "}";
+        return "WorkerGroup{" +
+                "id=" + id +
+                ", name='" + name + '\'' +
+                ", addrList='" + addrList + '\'' +
+                ", createTime=" + createTime +
+                ", updateTime=" + updateTime +
+                ", description='" + description + '\'' +
+                ", systemDefault=" + systemDefault +
+                '}';
```
Comment (ruanwenjun) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/WorkerGroup.java]: Please remove this and use Lombok.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java [hunk 2]
```diff
@@ -122,6 +123,7 @@ public Map<String, Object> saveWorkerGroup(User loginUser, int id, String name,
             return result;
         }
         if (workerGroup.getId() != 0) {
+            handleWorkGroup(workerGroup);
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java]: This is strange, I don't think we should add a handle method here, what the handle method means?

Comment (Zephyrcf) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java]: Persistent equal burial point

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/WorkerGroupController.java [hunk 3]
```diff
@@ -74,7 +74,8 @@ public class WorkerGroupController extends BaseController {
     @ApiImplicitParams({
         @ApiImplicitParam(name = "id", value = "WORKER_GROUP_ID", dataType = "Int", example = "10", defaultValue = "0"),
         @ApiImplicitParam(name = "name", value = "WORKER_GROUP_NAME", required = true, dataType = "String"),
-        @ApiImplicitParam(name = "addrList", value = "WORKER_ADDR_LIST", required = true, dataType = "String")
+        @ApiImplicitParam(name = "addrList", value = "WORKER_ADDR_LIST", required = true, dataType = "String"),
+        @ApiImplicitParam(name = "description", value = "WORKER_DESC", required = true, dataType = "String")
```
Comment (lenboo) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/WorkerGroupController.java]: required=false

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java [hunk 4]
```diff
@@ -304,6 +310,8 @@ private List<WorkerGroup> getWorkerGroups(boolean isPaging, List<Integer> ids) {
             if (childrenNodes == null || childrenNodes.isEmpty()) {
                 continue;
             }
+            WorkerGroup aDefault = workerGroups.stream().filter(a -> a.getAddrList().equals("default")).collect(Collectors.toList()).get(0);
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java]: This operation is not safe, I am not sure if the below code will return a empty list.
```
workerGroups.stream().filter(a -> a.getAddrList().equals("default")).collect(Collectors.toList())
```

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java (hunk 4 above)
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java]:
```suggestion
            WorkerGroup aDefault = workerGroups.stream().filter(a -> a.getAddrList().equals("default")).collect(Collectors.toList()).get(0);
```
Please put this out of the for loop

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java [hunk 5]
```diff
@@ -292,7 +298,10 @@ private List<WorkerGroup> getWorkerGroups(boolean isPaging, List<Integer> ids) {
             }
             return workerGroups;
         }
-
+        Map<String, WorkerGroup> collect = null;
+        if (workerGroups.size() != 0) { 
+            collect = workerGroups.stream().collect(Collectors.toMap(WorkerGroup::getName, a -> a, (k1, k2) -> k1));
```
Comment (lenboo) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java]: a  ?  k1 ？ k2?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java [hunk 6]
```diff
@@ -292,7 +298,10 @@ private List<WorkerGroup> getWorkerGroups(boolean isPaging, List<Integer> ids) {
             }
             return workerGroups;
         }
-
+        Map<String, WorkerGroup> workerGroupsMap = null;
+        if (workerGroups.size() != 0) {
+            workerGroupsMap = workerGroups.stream().collect(Collectors.toMap(WorkerGroup::getName, item -> item, (key, value) -> key));
```
Comment (lenboo) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java]: key ? value ?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java [hunk 7]
```diff
@@ -305,19 +314,27 @@ private List<WorkerGroup> getWorkerGroups(boolean isPaging, List<Integer> ids) {
...
+            handleAddrList(wg, workerGroup, childrenNodes);
             wg.setName(workerGroup);
             if (isPaging) {
-                wg.setAddrList(String.join(Constants.COMMA, childrenNodes));
                 String registeredValue = registryClient.get(workerGroupPath + Constants.SINGLE_SLASH + childrenNodes.iterator().next());
                 HeartBeat heartBeat = HeartBeat.decodeHeartBeat(registeredValue);
                 wg.setCreateTime(new Date(heartBeat.getStartupTime()));
                 wg.setUpdateTime(new Date(heartBeat.getReportTime()));
                 wg.setSystemDefault(true);
+                if (workerGroupsMap != null && workerGroupsMap.get(workerGroup) != null) {
```
Comment (lenboo) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java]: contains

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/registry/ServerNodeManager.java [hunk 8]
```diff
@@ -210,11 +210,7 @@ public void run() {
                         String workerGroup = wg.getName();
                         Set<String> nodes = new HashSet<>();
                         String[] addrs = wg.getAddrList().split(Constants.COMMA);
-                        for (String addr : addrs) {
-                            if (newWorkerNodeInfo.containsKey(addr)) {
-                                nodes.add(addr);
-                            }
-                        }
+                        handleAddr(newWorkerNodeInfo, workerGroup, nodes, addrs);
```
Comment (lenboo) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/registry/ServerNodeManager.java]: nodes = handleAddr(registerWorkerNodeMap, wg)

**Review by lenboo — DISMISSED** [review]
+1

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/registry/ServerNodeManager.java [hunk 9]
```diff
@@ -227,6 +221,16 @@ public void run() {
...
 
+
+    protected void handleAddr(Map<String, String> newWorkerNodeInfo, WorkerGroup wg, Set<String> nodes) {
+        String[] addrs = wg.getAddrList().split(Constants.COMMA);
+        for (String addr : addrs) {
+            if (newWorkerNodeInfo.containsKey(addr)) {
+                nodes.add(addr);
+            }
+        }
+    }
```
Comment (ruanwenjun) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/registry/ServerNodeManager.java]: it's better to rename this method to `protected Set<String> getNodeAddress(Map<String, String> newWorkerNodeInfo, WorkerGroup workgroup)`

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Pull request merged**